x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/transport/action/activate/ActivateWatchTests.java

```diff
@@ -107,11 +103,11 @@ public void testDeactivateAndActivate() throws Exception {
         assertThat(getWatchResponse, notNullValue());
         assertThat(getWatchResponse.getStatus().state().isActive(), is(true));
 
-        logger.info("Sleeping for another five seconds, ensuring that watch is executed");
-        Thread.sleep(5000);
         refresh();
-        long count3 = docCount(".watcher-history*", matchAllQuery());
-        assertThat(count3, greaterThan(count1));
+        assertBusy(() -> {
```

Comment:
I noticed you added the `5s` timeout to the first `awaitBusy()`, any reason it was not added here?

Reply:
there is a subtle difference...

the call above on line 93 is awaitBusy, which in the happy case always take 5 seconds, so this test will always take at least 5 seconds (ensuring nothing changed in that time period).

the call just above on line 107 is assertBusy, which in the happiest case will not need any waiting at all, but can wait up to 10 seconds if needed. It's a subtle change from the min time the test will take to minor increase in the max time the test _can_ take.

Reply:
yeesh, i didnt even see that. thanks for the explain.